controllers/server.js

```diff
@@ -30,6 +30,22 @@ const getBlogJS = () =>
     });
   });
 
+  
+const getPathQueryAndHash = (url) => {
+  const [path, rest] = url.split('?');
+  const [queryStr, hash] = (rest || '').split('#');
+  
+  const query = queryStr.split('&').map(part => {
+    const [key, value] = part.split('=');
+    return {[key]: value};
+  })
+  return {
+    pathname: path,
+    query: Object.fromEntries(query),
```

Comment:
in case of no params this will always pass an 
```
{undefined: undefined}
```
parameter, 
as well as an `#undefined` hash

probably better to clean this at some point...